[BE][inductor] tl.dot(..., allow_tf32=...) -> tl.dot(..., input_precision=...)

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #160711

allow_tf32 is deprecated. Also, this will make it easier to support tf32x3 (i.e. #160359).

dashboard results on h100 show no change: [inference](https://hud.pytorch.org/benchmark/compilers?dashboard=torchinductor&startTime=Mon%2C%2011%20Aug%202025%2017%3A01%3A22%20GMT&stopTime=Mon%2C%2018%20Aug%202025%2017%3A01%3A22%20GMT&granularity=hour&mode=inference&dtype=bfloat16&deviceName=cuda%20(h100)&lBranch=gh/davidberard98/399/orig&lCommit=ce12d0fd751a733f22b5bdda00bd58d323e0a526&rBranch=main&rCommit=e444cd24d48b3a46f067974f2cc157f5ed27709f), [training](https://hud.pytorch.org/benchmark/compilers?dashboard=torchinductor&startTime=Mon%2C%2011%20Aug%202025%2017%3A01%3A22%20GMT&stopTime=Mon%2C%2018%20Aug%202025%2017%3A01%3A22%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(h100)&lBranch=gh/davidberard98/399/orig&lCommit=ce12d0fd751a733f22b5bdda00bd58d323e0a526&rBranch=main&rCommit=e444cd24d48b3a46f067974f2cc157f5ed27709f)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @mlazos @chenyang78

davidberard98 committed:
[BE][inductor] tl.dot(..., allow_tf32=...) -> tl.dot(..., input_precision=...)

allow_tf32 is deprecated. Also, this will make it easier to support tf32x3.

[ghstack-poisoned]

davidberard98 committed:
Update on "[BE][inductor] tl.dot(..., allow_tf32=...) -> tl.dot(..., input_precision=...)"


allow_tf32 is deprecated. Also, this will make it easier to support tf32x3.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

PaulZhang12 reviewed (approved):
LGTM! Maybe also add an inductor dashboard e2e max-autotune run here?

davidberard98 committed:
rebase on 211c9885 to do a h100 max autotune run on "[BE][inductor] tl.dot(..., allow_tf32=...) -> tl.dot(..., input_precision=...)"


allow_tf32 is deprecated. Also, this will make it easier to support tf32x3 (i.e. #160359).

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben mlazos

[ghstack-poisoned]

davidberard98 committed:
Update on "[BE][inductor] tl.dot(..., allow_tf32=...) -> tl.dot(..., input_precision=...)"


allow_tf32 is deprecated. Also, this will make it easier to support tf32x3 (i.e. #160359).

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben mlazos

[ghstack-poisoned]

davidberard98 commented:
@pytorchbot merge

davidberard98 commented:
@pytorchbot revert -m "internal failure - T235384144 - I'll revert while I investigate."

exclamaforte commented:
@davidberard98 I haven't seen the internal failure, but I noticed that this PR doesn't respect the setting
```
torch.set_float32_matmul_precision("high")
```
which was causing the test_max_autotune tests to use tf32, for example

cw-tan commented:
@davidberard98 wondering if there has been any news about this PR and the associated Meta-internal failures?

davidberard98 commented:
Sorry, I'm not able to check.

cc @PaulZhang12 @NikhilAPatel in case you have bandwidth
